docs/developer-guide/get-started/building-howto.md

```diff
@@ -122,11 +122,123 @@ Then, rebuild the image:
+To add a new package you need to generate a `SPEC` file for the package which
+contains all required information for the build infrastructure to generate the
+`SRPM` and `RPM` for the package. There are a few steps involved to create
+a new package for the EMT.
 
-1. If a new package has to be released, follow these steps to ensure the package is available
-   in the artifactory:
+1. Defining the SPEC file and add it into the `/SPECS` directory
+1. Create the source archive and generate the sha256sum for the package
+1. Uploading the tar.gz package to the source package repository
+1. Add the SPEC file /SPEC directory and updating the `cgmanifest.json` file
+
+You need to first install the required build tools for `rpm`. On Fedora you
+can simply install the required packages with:
+
+```bash
+sudo dnf install rpm-build rpmdevtools
+rpmdev-setuptree
+```
+
+`rpmdev-setuptree` creates the necessary directories, which you may need to
+manually create on an Ubuntu distribution.
+
+```bash
+sudo apt-get install rpm
+mkdir -p ~/rpmbuild/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+echo '%_topdir %(echo $HOME)/rpmbuild' > ~/.rpmmacros
+```
+
+**Defining the SPEC file**
+Define and test that the SPEC file works as expected and is locally generating
+the required artifacts. This example builds a simple hello world `rpm` package
+which contains a bash scripts that prints hello world.
+
+```bash
+Name:           helloworld
+Version:        1.0
+Release:        1%{?dist}
+Summary:        Simple Hello World script
+
+License:        MIT
+URL:            https://example.com/helloworld
+Source0:        helloworld-1.0.tar.gz
+
+BuildArch:      noarch
+
+%description
+A very basic "Hello World" script packaged as an RPM.
+
+%prep
+%setup -q
+
+%build
+# Nothing to build for a shell script
+
+%install
+mkdir -p %{buildroot}/usr/bin
+install -m 0755 helloworld.sh %{buildroot}/usr/bin/helloworld
+
+mkdir -p %{buildroot}/usr/share/helloworld
+install -m 0644 helloworld.signature.json %{buildroot}/usr/share/helloworld/
+
+%files
+/usr/bin/helloworld
+/usr/share/helloworld/helloworld.signature.json
+
+%changelog
+* Wed May 01 2025 Your Name <you@example.com> - 1.0-1
+- Initial package
+```
+
+**Create the archive**: Create your source archive, add the simple script and
+make it executable.
+
+```bash
+mkdir helloworld-1.0
+
+cat <<EOF > ./helloworld-1.0/helloworld.sh
+#!/bin/bash
+echo "Hello, world!"
+EOF
+
+chmod +x ./helloworld-1.0/helloworld.sh
+
+sha256sum helloworld-1.0/helloworld.sh | awk '{print $1}' > sum.txt
+read sum < sum.txt
+echo "{\"file\": \"helloworld.sh\", \"sha256\": \"$sum\"}" > helloworld-1.0/helloworld.signature.json
+
+tar -czf helloworld-1.0.tar.gz helloworld-1.0/
+```
+
+Copy and build the `rpm` package to the rpm building directory.
+
+```bash
+cp helloworld-1.0.tar.gz ~/rpmbuild/SOURCES/
+cp helloworld.spec ~/rpmbuild/SPECS/
+rpmbuild -ba ~/rpmbuild/SPECS/helloworld.spec
+```
+
+**Uploading the archive**: Upload the tar.gz archive to [TBD].
```

Comment:
Can we move this to the bottom after image generation and testing? Add something like:

If testing is complete and you are ready to contribute this package, please raise a PR and work with a code owner to upload the source tarball to package source mirror.